storage/scylla_api.c

```diff
@@ -0,0 +1,707 @@
+  if (*transaction == NULL) {
+    ta_log_error("%s\n", "malloc failure");
+    ret = SC_STORAGE_OOM;
+    goto end_iterate;
+  }
+  /* get each transaction result */
+  idx = 0;
+  iterator = cass_iterator_from_result(result);
+  while (cass_iterator_next(iterator)) {
+    const CassRow* row = cass_iterator_get_row(iterator);
+
+    if ((ret = get_blob(row, ((*transaction) + idx)->bundle, "bundle")) != SC_OK) {
+      goto end_iterate;
+    }
+    if ((ret = get_blob(row, ((*transaction) + idx)->address, "address")) != SC_OK) {
+      goto end_iterate;
+    }
+    if ((ret = get_blob(row, ((*transaction) + idx)->hash, "hash")) != SC_OK) {
+      goto end_iterate;
+    }
+    if ((ret = get_blob(row, ((*transaction) + idx)->message, "message")) != SC_OK) {
+      goto end_iterate;
+    }
+    if ((ret = get_blob(row, ((*transaction) + idx)->trunk, "trunk")) != SC_OK) {
+      goto end_iterate;
+    }
+    if ((ret = get_blob(row, ((*transaction) + idx)->branch, "branch")) != SC_OK) {
+      goto end_iterate;
+    }
+    cass_value_get_int64(cass_row_get_column_by_name(row, "value"), &(((*transaction) + idx)->value));
+    cass_value_get_int64(cass_row_get_column_by_name(row, "timestamp"), &(((*transaction) + idx)->timestamp));
+
+    idx++;
+  }
+
+end_iterate:
+  cass_result_free(result);
+  cass_iterator_free(iterator);
+
+exit:
+  cass_future_free(future);
+  cass_statement_free(statement);
+
+  return ret;
+}
+
+static status_t push_columns_into_queue(CassSession* session, CassStatement* statement,
+                                        hash243_queue_t* const res_queue, const char* column_name) {
+  status_t ret = SC_OK;
+  CassFuture* future = NULL;
+  const CassResult* result;
+  CassIterator* iterator;
+
+  future = cass_session_execute(session, statement);
+  cass_future_wait(future);
+
+  if (cass_future_error_code(future) != CASS_OK) {
+    print_error(future);
+    ret = SC_STORAGE_CASSANDRA_QUREY_FAIL;
+    goto exit;
+  }
+
+  result = cass_future_get_result(future);
+  iterator = cass_iterator_from_result(result);
+
+  while (cass_iterator_next(iterator)) {
+    const CassRow* row = cass_iterator_get_row(iterator);
+    size_t len;
+    const cass_byte_t* buf;
+    if (cass_value_get_bytes(cass_row_get_column_by_name(row, column_name), &buf, &len) != CASS_OK) {
+      ret = SC_STORAGE_CASSANDRA_QUREY_FAIL;
+      goto end_iterate;
+    }
+    if (hash243_queue_push(res_queue, (flex_trit_t const* const)buf) != RC_OK) {
+      ret = SC_STORAGE_OOM;
+      goto end_iterate;
+    }
+  }
+
+end_iterate:
+  cass_result_free(result);
+  cass_iterator_free(iterator);
+
+exit:
+  cass_future_free(future);
+  cass_statement_free(statement);
+
+  return ret;
+}
+
+static void free_transaction(scylla_iota_transaction_t** trans) {
+  free((*trans)->address);
+  free((*trans)->branch);
+  free((*trans)->bundle);
+  free((*trans)->message);
+  free((*trans)->hash);
+  free((*trans)->trunk);
+  free(*trans);
```

Comment:
How about setting `trans = NULL`. Without setting it to NULL, actually you don't need to use pointer of pointer as parameter.

Reply:
BTW we used `txn` as abbreviation of transaction more often